airbyte-integrations/connectors/source-sentry/unit_tests/integration/test_issues.py

```diff
@@ -0,0 +1,300 @@
+        assert len(output.records) >= 1, f"Expected at least 1 record, got {len(output.records)}"
+
+        # ASSERT - Verify record content
+        record = output.records[0].record.data
+        assert record["id"] == "issue123", f"Expected issue123, got {record['id']}"
+        assert record["lastSeen"] == "2024-01-20T15:00:00Z", f"Expected lastSeen 2024-01-20, got {record['lastSeen']}"
+        assert record["title"] == "Cannot read property error"
+
+        # ASSERT - State message with latest lastSeen
+        assert len(output.state_messages) > 0, "Expected state messages to be emitted"
+        new_state = output.most_recent_state.stream_state.__dict__
+        # State should be updated to the latest record (01-20)
+        assert new_state["lastSeen"].startswith("2024-01-20T15:00:00"), f"Expected state to advance to latest record, got {new_state}"
+
+    @HttpMocker()
+    def test_incremental_sync_first_sync_emits_state(self, http_mocker: HttpMocker):
+        """
+        Test first incremental sync with no previous state for issues stream.
+
+        This tests:
+        - Connector works in incremental mode without existing state (first sync)
+        - Records are returned using default behavior (no start time filtering)
+        - State message is emitted with latest record's lastSeen
+        """
+        # ARRANGE - Mock API returns issues (no state, so uses default behavior)
+        http_mocker.get(
+            SentryRequestBuilder.issues_endpoint(_ORGANIZATION, _PROJECT, _AUTH_TOKEN).build(), create_response("issues", has_next=False)
+        )
+
+        # ACT - First incremental sync (no state parameter)
+        source = get_source(config=self._config())
+        catalog = CatalogBuilder().with_stream(_STREAM_NAME, SyncMode.incremental).build()
+        output = read(source, config=self._config(), catalog=catalog)
+
+        # ASSERT - Records returned
+        assert len(output.records) >= 1, f"Expected at least 1 issue record, got {len(output.records)}"
+
+        # ASSERT - Record values
+        first_record = output.records[0].record.data
+        assert first_record["id"] == "issue123", f"Expected issue123, got {first_record['id']}"
+        assert first_record["lastSeen"] == "2024-01-20T15:00:00Z", f"Expected lastSeen timestamp, got {first_record['lastSeen']}"
+
+        # ASSERT - State message emitted with cursor value (KEY VALIDATION)
+        assert len(output.state_messages) > 0, "Expected state messages to be emitted on first sync"
+        state = output.most_recent_state.stream_state.__dict__
+        # State should be set to the latest record's lastSeen
+        assert state["lastSeen"] is not None, "Expected state to have lastSeen cursor"
+        assert state["lastSeen"].startswith("2024-01-20T15:00:00"), f"Expected state cursor to be latest record's lastSeen, got {state}"
+
+    @HttpMocker()
+    def test_incremental_pagination_with_smart_stopping(self, http_mocker: HttpMocker):
```

Comment:
nit: I don't really like this name `with_smart_stopping`. It feels very vague and ambiguous. We should call this `with_data_feed` or something like that because that's the CDK/sentry feature that we're actually testing